1-JavaScript/Week3/homework/ex2-mondaysWorth.test.js

```diff
@@ -31,15 +20,29 @@ const mondayTasks = [
 
 const hourlyRate = 25;
 
-function computeEarnings(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function computeEarnings(tasks , hourlyRate) {
+  if (
+    !Array.isArray(tasks) ||
+    tasks.every(
+      (task) =>
+        typeof task !== 'object' ||
+        !('duration' in task) ||
+        typeof task.duration !== 'number')){
+    return 'invalid input';
+  } else {
+    const durationInHours = tasks.map((task) => task.duration / 60);
+    const billingSum = durationInHours.reduce(
+      (accumulator, duration) => accumulator + duration * hourlyRate,
+      0
+    );
+    return `€${billingSum.toFixed(2)}`;
+  }
```

Comment:
Wow, nice work here! 👍 

Not sure if you need to check so extensively if certain things are present in the array, but it's good to be mindful of things going wrong. 

I like how you named it `durationInHours`, super clean and clear! Also the use of `.reduce` is great!

One tip to possibly improve this: you can chain `.map` and `.reduce` together! It's even possible to do this exercise with one line using only `.reduce` 😲 But that's an extra challenge if you like it, not necessary.

```suggestion
function computeEarnings(tasks , hourlyRate) {
  if (
    !Array.isArray(tasks) ||
    tasks.every(
      (task) =>
        typeof task !== 'object' ||
        !('duration' in task) ||
        typeof task.duration !== 'number')){
    return 'invalid input';
  } else {
    const durationInHours = tasks.map((task) => task.duration / 60).reduce(
      (accumulator, duration) => accumulator + duration * hourlyRate,
      0
    );
    return `€${billingSum.toFixed(2)}`;
  }
```